shim/batch.go

```diff
@@ -0,0 +1,69 @@
+// Copyright the Hyperledger Fabric contributors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package shim
+
+import (
+	"github.com/hyperledger/fabric-protos-go-apiv2/peer"
+)
+
+type writeBatch struct {
+	writes map[string]*peer.WriteRecord
+}
+
+func newWriteBatch() *writeBatch {
+	return &writeBatch{
+		writes: make(map[string]*peer.WriteRecord),
+	}
+}
+
+func (b *writeBatch) Writes() []*peer.WriteRecord {
+	if b == nil {
+		return nil
+	}
+
+	var results []*peer.WriteRecord
+	for _, value := range b.writes {
+		results = append(results, value)
+	}
+
+	return results
+}
```

Comment:
Why are you converting map to slice?
I think it's an unnecessary action.

Reply:
I suggest using the library "golang.org/x/exp/maps" ver
[v0.0.0-20240222234643-814bf88cf225](https://pkg.go.dev/golang.org/x/exp@v0.0.0-20240222234643-814bf88cf225/maps)

and the code will look like

```
return maps.Values(b.writes)

..............................

for v := range s.writeBatch. Writes() {
	...........
}
```

Reply:
Your code also converted the map values to a slice. It just did it in a different place.

Reply:
What if I hand over the map?
My goal is to increase the speed of transaction processing.
Maybe the extra transfer will affect the speed.

Reply:
We should not use experimental libraries. Standard library feature for getting map values is in Go 1.23. Unfortunately we are at Go 1.22

Reply:
Okay. We forgot about the experimental libraries.
Could you do benchmark tests and check how much time we lose on moving from map to slice? 1000 values? 10000 values?